Remotion/Data/DomainObjects.UnitTests/Persistence/Rdbms/Parameters/SerializedObjectIDDataParameterDefinitionTest.cs

```diff
@@ -0,0 +1,145 @@
+
+namespace Remotion.Data.DomainObjects.UnitTests.Persistence.Rdbms.Parameters;
+
+[TestFixture]
+public class SerializedObjectIDDataParameterDefinitionTest : StandardMappingTest
+{
+  [Test]
+  public void Initialize ()
+  {
+    var storageTypeInformation = StorageTypeInformationObjectMother.CreateStorageTypeInformation();
+
+    var parameterDefinition = new SerializedObjectIDDataParameterDefinition(storageTypeInformation);
+
+    Assert.That(parameterDefinition.StorageTypeInformation, Is.SameAs(storageTypeInformation));
+  }
+
+  [Test]
+  public void Initialize_WithIncorrectStorageTypeInformation ()
+  {
+    var storageTypeInformation = StorageTypeInformationObjectMother.CreateIntStorageTypeInformation();
+
+    Assert.That(
+        () => new SerializedObjectIDDataParameterDefinition(storageTypeInformation),
+        Throws.InstanceOf<ArgumentException>().With.ArgumentExceptionMessageEqualTo(
+            string.Format("Parameter '{0}' has type '{2}' when type '{1}' was expected.", "storageTypeInformation.DotNetType", typeof(string), typeof(int)),
+            "storageTypeInformation.DotNetType"));
+  }
+
+  [Test]
+  public void GetParameterValue ()
+  {
+    var storageTypeInformation = StorageTypeInformationObjectMother.CreateStorageTypeInformation(storageDbType: DbType.AnsiString, storageTypeLength: 255);
+
+    var objectID = DomainObjectIDs.Order1;
+    var expectedValue = ObjectIDStringSerializer.Instance.Serialize(objectID);
+
+    var serializedObjectIDDataParameterDefinition = new SerializedObjectIDDataParameterDefinition(storageTypeInformation);
+
+    var result = serializedObjectIDDataParameterDefinition.GetParameterValue(objectID);
+
+    Assert.That(result, Is.EqualTo(expectedValue));
+  }
+
+  [Test]
+  public void GetParameterValue_WithNull_ReturnsDBNull ()
+  {
+    var storageTypeInformation = StorageTypeInformationObjectMother.CreateStorageTypeInformation();
+
+    var serializedObjectIDDataParameterDefinition = new SerializedObjectIDDataParameterDefinition(storageTypeInformation);
+
+    Assert.That(serializedObjectIDDataParameterDefinition.GetParameterValue(null), Is.EqualTo(DBNull.Value));
+  }
+
+  [Test]
+  public void GetParameterValue_WithTypeMismatchedValue_ThrowsArgumentException ()
+  {
+    var dummyValue = "Not an ObjectID";
+    var storageTypeInformation = StorageTypeInformationObjectMother.CreateStorageTypeInformation();
+
+    var serializedObjectIDDataParameterDefinition = new SerializedObjectIDDataParameterDefinition(storageTypeInformation);
+
+    Assert.That(
+        () => serializedObjectIDDataParameterDefinition.GetParameterValue(dummyValue),
+        Throws.InstanceOf<ArgumentException>().With.ArgumentExceptionMessageEqualTo(
+            $"Parameter 'value' has type '{typeof(string)}' when type '{typeof(ObjectID)}' was expected.",
+            "value"));
+  }
+
+  [Test]
+  public void CreateDataParameter ()
+  {
+    var commandMock = new Mock<IDbCommand>(MockBehavior.Strict);
```

Comment:
```suggestion
    var commandMock = new Mock<IDbCommand>();
```
Do we need this as a strict mock or could it simply be a stub?